migration/migrations/course/20180611125006_alphanumeric_sections.py

```diff
@@ -0,0 +1,40 @@
+def up(conn):
+    with conn.cursor() as cursor:
+        # drop foreign key contraints as the foreign keys also store type of columns in their definitions
+        cursor.execute("ALTER TABLE ONLY grading_registration DROP CONSTRAINT grading_registration_sections_registration_id_fkey")
+        cursor.execute("ALTER TABLE ONLY users DROP CONSTRAINT users_registration_section_fkey")
+
+        # typecast from integer to character varying
+        cursor.execute("ALTER TABLE ONLY sections_registration ALTER COLUMN sections_registration_id SET DATA TYPE character varying(255) USING sections_registration_id::varchar(255)")
+        cursor.execute("ALTER TABLE ONLY grading_registration ALTER COLUMN sections_registration_id SET DATA TYPE character varying(255) USING sections_registration_id::varchar(255)")
+        cursor.execute("ALTER TABLE ONLY users ALTER COLUMN registration_section SET DATA TYPE character varying(255) USING registration_section::varchar(255)")
+        cursor.execute("ALTER TABLE ONLY gradeable_teams ALTER COLUMN registration_section SET DATA TYPE character varying(255) USING registration_section::varchar(255)")
+
+        # renable the foreign key contraints as we're done typecasting
+        cursor.execute("ALTER TABLE ONLY grading_registration ADD CONSTRAINT grading_registration_sections_registration_id_fkey FOREIGN KEY (sections_registration_id) REFERENCES sections_registration(sections_registration_id)")
+        cursor.execute("ALTER TABLE ONLY users ADD CONSTRAINT users_registration_section_fkey FOREIGN KEY (registration_section) REFERENCES sections_registration(sections_registration_id)")
+
+        # add missing foreign key contraints
+        cursor.execute("ALTER TABLE ONLY gradeable_teams ADD CONSTRAINT gradeable_teams_registration_section_fkey FOREIGN KEY (registration_section) REFERENCES sections_registration(sections_registration_id)")
+        cursor.execute("ALTER TABLE ONLY gradeable_teams ADD CONSTRAINT gradeable_teams_rotating_section_fkey FOREIGN KEY (rotating_section) REFERENCES sections_rotating(sections_rotating_id)")
+
+
+def down(conn):
+    with conn.cursor() as cursor:
+        # drop foreign key contraints while we typecast the columns
+        cursor.execute("ALTER TABLE ONLY gradeable_teams DROP CONSTRAINT gradeable_teams_registration_section_fkey")
+        cursor.execute("ALTER TABLE ONLY gradeable_teams DROP CONSTRAINT gradeable_teams_rotating_section_fkey")
+
+        # disable foreign key contraints while we typecast the columns
+        cursor.execute("ALTER TABLE ONLY grading_registration DROP CONSTRAINT grading_registration_sections_registration_id_fkey")
+        cursor.execute("ALTER TABLE ONLY users DROP CONSTRAINT users_registration_section_fkey")
+
+        # typecast from integer to character varying
```

Comment:
@MasterOdin 

Don't you mean from character to integer?

Also...  What will happen if the section isn't an integer?  How will it fail?   We had discussed simply deleting that section and moving all references to that to be null, but that's actually messy & complicated & unnecessary.  Let's just document it really clearly in the PR message.

Reply:
yeah, I was just copy-pasting the sections from the up method to the down method and then modifying the contents. Whoops.

If the sections are not integers and you try to run down, this will throw an error giving something like `ERROR:  invalid input syntax for integer: "a"`. I'd agree with just documenting that this migration is non-reversible if you change any section to a non-numeric. However, it should be that most people never need to run the down method so I wouldn't spend any time worrying about this case beyond just documenting it.